### DISCUSSION TITLE
Data loader reloads on navigating to child routes?

### DISCUSSION THREAD

**Discussion by ricktibbe** [posva/unplugin-vue-router#537] [Questions and Help] [answered]
I've got the following pages / routes in my Vue app:

1. `/clients/[clientId]`
2. `/clients/[clientId]/orders/[orderId]`

Page 1 consists of 2 columns: on the left, the general information of the client is shown, which is loaded through the `useClient()` loader. The right column is a child `<RouterView>`, which shows an overview of the client's orders on page 1 (`useOrders()`), and a specific order on page 2 (`useCurrentOrder()`).

The general client information only has to be loaded once, as soon as the user navigates to page 1. However, as soon as the user navigates from page 1 to page 2, I noticed that the `useClient()` data loader is being reloaded, as it fetches all client data again. As the client's data does not change in this case, I actually do not want the `useClient()` loader to reload. How can I solve this?

---

My route structure looks like this:
```
Pages/
└── clients/
    ├── [clientId]/
    │   ├── orders/
    │   │   └── [orderId].vue
    │   └── index.vue
    └── [clientId].vue 
```

`useClient()` is exported in `[clientId].vue`, whereas `useOrders()` is exported in `[clientId]/index.vue`.

**Comment by ricktibbe** [comment]
@posva do you know if this is a bug, or intended behavior?

**Reply by posva** [reply, marked as answer]
This sounds like intended behavior. With the colada loader, it does track the properties that you consume on the route so it won’t trigger again but not with the basic one

**Reply by posva** [reply]
That being said, it might be better not to trigger loaders by default when navigating between nested routes. Or an option to configure it
